### PR TITLE
Implement publish intent

### PR DESCRIPTION
**Pull request by GhostofCookie**
Getting some implementation added for publish intent. Tried not to modify existing stuff too much (trying to honour the intent of the implementation (I think)).

Some other changes to ancillary stuff like MessageBuffer and Name and Namespace are just little optimizations or code cleanup.

### CODE REVIEW

**Comment by GhostofCookie** [comment]
I'm not in love with the entirety of this implementation, I think I'm to mark it as draft and WIP and go back in and make the publish intent do something a little more meaningful.

**Comment by GhostofCookie** [comment]
Okay, now this actually does something a little more meaningful; Actually registers a namespace to publish to.

**Comment by GhostofCookie** [comment]
And now publish end is on here as well, so calling Remove stream will not scream anymore, and even cleanup a little.

File: include/quicr/quicr_common.h [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace quicr {
  *    end-to-end, 1400 - 119 = 1281.  A max data size of 1280 should be good
  *    end-to-end for all paths.
  */
-const uint16_t MAX_TRANSPORT_DATA_SIZE = 1280;
+constexpr uint16_t MAX_TRANSPORT_DATA_SIZE = 1280;
```
Comment (GhostofCookie) [inline include/quicr/quicr_common.h]: This conflicts with @TimEvens PR I think.

**Review by shenning00 — APPROVED** [review]
Plenty of changes. 

Will work to get some test code in place in qmedia.

**Pull request merged**